### PR TITLE
Nodejs

### PR DESCRIPTION
**Pull request by katwalmayank**
Most up to date node js branch with all proper methods. still needs unit testing

### CODE REVIEW

File: routes/user/index.js [hunk 1]
```diff
@@ -5,96 +5,103 @@ const path = require('path');
...
 
-//login in to check if id = email for account 
-router.post('/:id/email/:email',controller.checkEmail);
-
-
 //get email by id
 router.get("/:id/email", controller.getEmailById);
 //update email by id
-router.put("/:id/email/:email", controller.updateEmailById);
+router.put("/:id/email", controller.updateEmailById);
```
Comment (kmui2) [inline routes/user/index.js]: Why remove `:email` param?

File: config/helper.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+var crypto = require('crypto');
```
Comment (kmui2) [inline config/helper.js]: Use `const` instead of `var`.

File: routes/match/match.controller.js [hunk 3]
```diff
@@ -63,13 +64,64 @@ module.exports.getMatchById = (req, res) => {
...
+    if (err) {
+      throw err;
+    }
+    console.log(rows)
+    res.json({ success: true, rows: rows });
+  });
+  
+}
+
+function userIDType(id){
```
Comment (kmui2) [inline routes/match/match.controller.js]: Add comments.

File: routes/user/index.js [hunk 4]
```diff
@@ -5,96 +5,103 @@ const path = require('path');
...
 
-//CREATE all five tables
-router.post('/tables',controller.createTables);
+//CREATE all 7 tables
+router.post('/tables', controller.createTables);
 
-//delete table
-router.delete('/tables',controller.deletetable);
+//Delete all 7 tables
+router.delete('/tables', controller.deletetable);
```
Comment (kmui2) [inline routes/user/index.js]: Camel case preferred for `deleteTable`.

File: routes/user/user.controller.js [hunk 5]
```diff
@@ -3,23 +3,26 @@ const path = require("path");
 const fs = require('fs');
 const config = require("../../config/database.js");
 const db = require('../../db');
+var date = new Date();
```
Comment (kmui2) [inline routes/user/user.controller.js]: Use `const` instead of `var`.

File: routes/user/user.controller.js [hunk 6]
```diff
@@ -3,23 +3,26 @@ const path = require("path");
...
+
 const os = require('os');
 const ip = require('ip');
 const { getRandomArbitrary } = require('../../utils/getRandomArbitrary');
 
 
 var user_sql_constants = require("../../config/user_sql_constants.js");
+var hp = require("../../config/helper.js");
```
Comment (kmui2) [inline routes/user/user.controller.js]: Use const instead of `var`.

File: routes/user/user.controller.js [hunk 7]
```diff
@@ -3,23 +3,26 @@ const path = require("path");
...
 
 //create all tables
 module.exports.createTables = (req, res) => {
-  
+
   sql1 = user_sql_constants.create_mentor_table_sql();
```
Comment (kmui2) [inline routes/user/user.controller.js]: Define all variables using `const`.

File: routes/user/user.controller.js [hunk 8]
```diff
@@ -50,17 +53,24 @@ module.exports.createTables = (req, res) => {
...
+      throw err;
+    }
+  });
+  res.json({ success: true, rows: "all tables created" });
 }
 //drop table 
 module.exports.deletetable = (req, res) => {
-  
+
   sql1 = `DROP TABLE IF EXISTS Mentors;`;
```
Comment (kmui2) [inline routes/user/user.controller.js]: Define all variables using `const`.

File: routes/user/user.controller.js [hunk 9]
```diff
@@ -81,48 +91,240 @@ module.exports.deletetable = (req, res) => {
...
-  sql = user_sql_constants.post_mentor_sql(user);
-  console.log(sql);
-  db.all(sql, [], (err, rows) => {
-  if (err) {
-    throw err;
-  }
-  res.json({ success: true, rows: rows });
-});
+
+    date = new Date()
```
Comment (kmui2) [inline routes/user/user.controller.js]: Define all variables using `const`.

File: routes/user/user.controller.js [hunk 10]
```diff
@@ -81,48 +91,240 @@ module.exports.deletetable = (req, res) => {
...
+  });
 }
 //create new mentee
 module.exports.postMentee = (req, res) => {
-  const fields = ['first_name', 'last_name', 'email_address' ,'biography','zipcode',
-  'date_of_birth','skills','hobbies'];
+  const fields = ['first_name', 'last_name', 'email_address', 'biography', 'zipcode',
+    'date_of_birth', 'skills', 'profession', 'profile_pic_URL', 'hobbies'];
+  const user = {};
+  fields.forEach(field => {
```
Comment (kmui2) [inline routes/user/user.controller.js]: Switch `forEach` with `some` and create variable `missingFields` similar to the following format:
https://github.com/RiseCareerAcademy/Rise/blob/6f285da2a2b7a00b01f4925c39e960ee1f9c2fcc/routes/user/user.controller.js#L94

File: routes/user/user.controller.js [hunk 11]
```diff
@@ -134,121 +336,118 @@ module.exports.postMentee = (req, res) => {
...
-  user.user_id = getRandomArbitrary(0, 100000);
-  ip_address = ip.address();
-  user.profile_pic_URL = `http://${ip_address}:8000/user/${user.user_id}/profilepic`;
-
-  sql = user_sql_constants.post_mentee_sql(user)
-  console.log(sql);
-  db.all(sql, [], (err, rows) => {
+  console.log(user)
+  sql = `INSERT INTO Messages VALUES ('${date.getTime()}',?,?,?,?,?)`
+  console.log(date.getTime())
```
Comment (kmui2) [inline routes/user/user.controller.js]: Remove all console logs.

File: config/helper.js (hunk 2 above)
Comment (katwalmayank) [reply config/helper.js]: in the body

Comment (kmui2) [reply config/helper.js]: What?

File: routes/user/index.js (hunk 1 above)
Comment (katwalmayank) [reply routes/user/index.js]: in the body

File: routes/user/index.js (hunk 4 above)
Comment (katwalmayank) [reply routes/user/index.js]: what is camel case

**Pull request merged**